src/runner-esm.mjs

```diff
@@ -149,29 +154,32 @@ export default class MoleculerRunner {
 	async loadConfigFile() {
 		let filePath;
 		// Env vars have priority over the flags
-		if (process.env["MOLECULER_CONFIG"]) {
-			filePath = path.isAbsolute(process.env["MOLECULER_CONFIG"])
-				? process.env["MOLECULER_CONFIG"]
-				: path.resolve(process.cwd(), process.env["MOLECULER_CONFIG"]);
-		} else if (this.flags.config) {
-			filePath = path.isAbsolute(this.flags.config)
-				? this.flags.config
-				: path.resolve(process.cwd(), this.flags.config);
-		}
-		if (!filePath && fs.existsSync(path.resolve(process.cwd(), "moleculer.config.mjs"))) {
-			filePath = path.resolve(process.cwd(), "moleculer.config.mjs");
+		const configPath = process.env["MOLECULER_CONFIG"] || this.flags.config;
+
+		if (configPath != null) {
+			if (path.isAbsolute(configPath)) {
+				filePath = this.tryConfigPath(configPath);
+			} else {
+				filePath = this.tryConfigPath(path.resolve(process.cwd(), configPath));
+
+				if (filePath == null) {
+					filePath = this.tryConfigPath(configPath, true);
+				}
+			}
+
+			if (filePath == null) {
+				return Promise.reject(new Error(`Config file not found: ${configPath}`));
+			}
 		}
-		if (!filePath && fs.existsSync(path.resolve(process.cwd(), "moleculer.config.js"))) {
-			filePath = path.resolve(process.cwd(), "moleculer.config.js");
+
+		if (filePath == null) {
```

Comment:
The original `runner-esm` code tries to load `moleculer.config.mjs` as well.

Reply:
Good catch. I've added a check for `moleculer.config.mjs` here.